core/src/components/datetime-button/datetime-button.scss

```diff
@@ -18,6 +18,10 @@
 
   position: relative;
 
+  flex: 0 1 auto;
+
+  min-width: 0;
```

Comment:
What does this do?

Reply:
Actually I had this in the wrong place. It is necessary for when an `ion-datetime-button` is inside of an `ion-item` (so all of the `ion-datetime/basic` tests). It should be on the host, I updated it here: https://github.com/ionic-team/ionic-framework/pull/27848/commits/e52c28bcefe93370e9e81dd498b428d2161697d6

Also added a test here: https://github.com/ionic-team/ionic-framework/pull/27848/commits/8ea5fcd1c1b380d7e7b50e236c810a26fb5f3639

Here's how it looks without and with the `min-width: 0`:

| without `min-width: 0` | with `min-width: 0` |
| ---| ---|
| ![image 9](https://github.com/ionic-team/ionic-framework/assets/6577830/dafe99c5-9fc3-49d6-8277-dd69867cb5ba) | ![image 11](https://github.com/ionic-team/ionic-framework/assets/6577830/8d9bd846-fcb6-44cc-a54f-63649166087a) | 

I'm running screenshots now so this should be updated soon.